magda-web-client/src/reducers/datasetSearchReducer.js

```diff
@@ -46,13 +46,17 @@ const datasetSearchReducer = (
 ) => {
     switch (action.type) {
         case "REQUEST_RESULTS":
+            console.log(action.json);
             return Object.assign({}, state, {
                 isFetching: true,
                 error: null,
                 apiQuery: action.apiQuery && action.apiQuery,
-                temporalRange: initialData.temporalRange,
-                publisherOptions: initialData.publisherOptions,
-                formatOptions: initialData.formatOptions,
+                temporalRange: [
```

Comment:
(comment 3/3)

... so then here, `activePublishers`, `activeRegion` etc can be set to the values in the query object that you've passed through as `action.query`.

There's some wrinkles here - I'm not sure whether you still need the `apiQuery` string (what does it do? :/) and you might need to smooth over what happens with the filters if you've come to the search page for the first time and they don't have options yet but do have active values.